modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/prepare/IgniteSqlToRelConvertor.java

```diff
@@ -69,6 +83,98 @@ public IgniteSqlToRelConvertor(
         }
     }
 
+    @Override protected RelNode convertInsert(SqlInsert call) {
+        datasetStack.push(call);
+
+        RelNode rel = super.convertInsert(call);
+
+        datasetStack.pop();
+
+        return rel;
+    }
+
+    private static class DefaultChecker extends SqlShuttle {
+        @Override public @Nullable SqlNode visit(SqlCall call) {
+            if (call.getKind() == SqlKind.DEFAULT) {
+                throw new ControlFlowException();
+            }
+
+            return super.visit(call);
+        }
+    }
+
+    @Override public RelNode convertValues(SqlCall values, RelDataType targetRowType) {
+        boolean defaultValueFound = false;
+
+        DefaultChecker checker = new DefaultChecker();
+
+        try {
+            values.accept(checker);
+        } catch (ControlFlowException e) {
+            defaultValueFound = true;
+        }
+
+        if (defaultValueFound) {
+            SqlValidatorScope scope = validator.getOverScope(values);
+            assert scope != null;
+            Blackboard bb = createBlackboard(scope, null, false);
+
+            convertValuesImplEx(bb, values, targetRowType);
+            return bb.root();
+        } else {
+            // a bit lightweight than default processing one.
+            return super.convertValues(values, targetRowType);
+        }
+    }
+
+    private void convertValuesImplEx(Blackboard bb, SqlCall values, RelDataType targetRowType) {
+        SqlCall insertOp = datasetStack.peek();
+        assert insertOp instanceof SqlInsert;
+        assert values == ((SqlInsert) insertOp).getSource();
+        RelOptTable targetTable = getTargetTable(insertOp);
+        assert targetTable != null;
+
+        IgniteTable ignTable = targetTable.unwrap(IgniteTable.class);
+
+        List<RelDataTypeField> tblFields = targetTable.getRowType().getFieldList();
+        List<String> targetFields = targetRowType.getFieldNames();
+
+        for (SqlNode rowConstructor : values.getOperandList()) {
+            SqlCall rowConstructor0 = (SqlCall) rowConstructor;
+
+            List<Pair<RexNode, String>> exps = new ArrayList<>(targetFields.size());
+
+            int pos = 0;
+            int rowPos = 0;
+            for (RelDataTypeField fld : tblFields) {
+                if (fld.getName().equals(targetFields.get(rowPos))) {
+                    SqlNode operand = rowConstructor0.getOperandList().get(rowPos);
+
+                    if (operand.getKind() == SqlKind.DEFAULT) {
```

Comment:
Can operand be null?
Is 3-rd value allowed:  INSERT INTO t VALUES (1,100), (2, DEFAULT), (3)

Reply:
Such kind of query format is not possible )